lib/src/lexer/lexer.dart

```diff
@@ -153,6 +153,60 @@ final class Lexer {
     }
   }
 
+  void _advanceNumericAndSeparatorMany1() {
+    // The first character must be numeric
+    if (_isNumeric(_peek)) {
+      _advance();
+    } else {
+      _errorHandler?.emit(
+        DecimalPartNotStartingWithANumberError(offset: _current),
+      );
+      return;
+    }
+
+    while ((_isNumeric(_peek) || _peek == '_') && !_isAtEnd) {
+      final isSeparator = _peek == '_';
+
+      _advance();
+
+      if (_peek == '_' && isSeparator) {
+        // two neighbour separators found (e.g. 14__4)
```

Comment:
Now I'm thinking, do we want to prevent multiple separators? IIRC some people use it to differentiate larger groups, and I think Dart separators specification consider this.

Example from [Dart's specification](https://github.com/dart-lang/language/blob/main/accepted/future-releases/digit-separators/feature-specification.md#examples):
```
100__000_000__000_000__000_000  // one hundred million million millions!
```

Let me know what you thinkg.

Reply:
It's actually easier to parse, too.